src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,272 @@
+     *         distance from the start edge's
+     */
+    static Predicate<Edge> getCandidateEdges(final Edge startEdge,
+            final int maximumAllowedDifferences)
+    {
+        // evaluate name tag of startEdge vs incoming Edge
+        return incomingEdge ->
+        {
+            final int similarityIndex = similarityIndex(incomingEdge, startEdge);
+            return similarityIndex <= maximumAllowedDifferences && similarityIndex > 0;
+        };
+    }
+
+    /**
+     * @param first
+     *            a character to examine for equality with second
+     * @param second
+     *            a character to examine for equality with first
+     * @return 0 if first & second are equal, 1 otherwise
+     */
+    private static int costOfSubstitution(final char first, final char second)
+    {
+        return first == second ? 0 : 1;
+    }
+
+    /**
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumSearchDistance
+     *            the maximum distance from the end of the incoming edge to the start of the
+     *            starting edge
+     * @return A stream of edges that fall in the search area
+     */
+    private static Function<Edge, Stream<Edge>> edgesWithinMaximumSearchDistance(
+            final Edge startEdge, final Distance maximumSearchDistance)
+    {
+        return incomingEdge -> incomingEdge.end().getLocation()
+                .distanceTo(startEdge.start().getLocation())
+                .isLessThanOrEqualTo(maximumSearchDistance)
+                        ? incomingEdge.connectedEdges().stream().filter(Edge::isMasterEdge)
+                        : Stream.empty();
+    }
+
+    /**
+     * @param evaluateEdgeCharacter
+     *            the character being checked for equality with the lambda parameter
+     * @return true if the incoming character is the same as the parameter character
+     */
+    private static Predicate<String> equalCharacter(final char evaluateEdgeCharacter)
+    {
+        return directionCharacter -> evaluateEdgeCharacter == getCharacter(directionCharacter);
+    }
+
+    /**
+     * @param oneCharacterString
+     *            the string to be converted to a character
+     * @return a char representation of the parameter String
+     */
+    private static char getCharacter(final String oneCharacterString)
+    {
+        return oneCharacterString.charAt(0);
+    }
+
+    /**
+     * Compute the Levenshtein distance between two road names. Code used to calculate Levenshtein
+     * distance is adapted from https://www.baeldung.com/java-levenshtein-distance
+     *
+     * @param incomingEdgeName
+     *            the name of the next edge in the search area
+     * @param startingEdgeName
+     *            the name of the edge from which the search started
+     * @return the Levenshtein distance between two edge's names
+     */
+    private static int getLevenshteinDistance(final String incomingEdgeName,
```

Comment:
There is an existing implementation of this available through on of the apache libraries: `org.apache.commons.lang3.StringUtils.getLevenshteinDistance()`. I'm not sure how useful it is given the need for handling the direction and number differences.

Reply:
Right, the main problem with that implementation is it doesn't handle direction and number differences. It simply takes in 2 strings and returns an integer. We need more granular control of the Levenshtein function to work with direction/number differences so I decided a custom Levenshtein method would do best. Also, ```StringUtils.getLevenshteinDistance()``` is deprecated.

Reply:
First of all, respect for this! Love to see dynamic programming in the wild 🙌

I would love to see this pulled out into a utility somewhere. In addition to making it public for others to use, that allows us to write unit tests for it which will increase our confidence in this covering all of the edge cases.

As another thought, do you see the threshold for this ever increasing above 1? Or does it not make sense to ever check for a minimum edit distance of 2, for example?